dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -448,12 +447,50 @@ HRESULT WINAPI CDesktopFolder::CompareIDs(LPARAM lParam, PCUIDLIST_RELATIVE pidl
 {
     bool bIsDesktopFolder1, bIsDesktopFolder2;
 
+    if (LOWORD(lParam) >= DESKTOPSHELLVIEWCOLUMNS)
+    {
+        ERR("Got invalid column pointer %Ix\n", LOWORD(lParam));
+        return E_INVALIDARG;
+    }
+
     if (!pidl1 || !pidl2)
     {
         ERR("Got null pidl pointer (%Ix %p %p)!\n", lParam, pidl1, pidl2);
         return E_INVALIDARG;
     }
 
+    if (lParam == 3) // file modified date from index into DesktopSFHeader[]
```

Comment:
Everywhere, would it be overkill to use `DesktopSFHeader[lParam] == IDS_SHV_COLUMN_MODIFIED` pattern?

Reply:
I used for form of this suggestion in a few places for now. We will see how the reviews respond. Thanks.

Reply:
The code has been reworked to simplify the code and eliminate the need for testing like this.